Tests/AsyncHTTPClientTests/HTTPClientTestUtils.swift

```diff
@@ -490,6 +490,95 @@ extension ByteBuffer {
     }
 }
 
+/// An abstract class that makes building simple asynchronous operations easy.
+/// Subclasses must implement `execute()` to perform any work and call
+/// `finish()` when they are done. All `NSOperation` work will be handled
+/// automatically.
+open class AsynchronousOperation: Operation {
+
+    // MARK: - Properties
+    private let stateQueue = DispatchQueue(
+        label: "com.asynchttpclient.operation.state",
+        attributes: .concurrent)
+
+    private var rawState = OperationState.ready
+
+    @objc private dynamic var state: OperationState {
```

Comment:
can we remove all the `@objc` stuff? That won't work on Linux I don't think. Also is the KVO/KVC? That won't work either I don't think.